src/sentry/static/sentry/app/components/activity/item/index.jsx

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import TimeSince from 'app/components/timeSince';
+import space from 'app/styles/space';
+import textStyles from 'app/styles/text';
+
+import ActivityAvatar from './avatar';
+import ActivityBubble from './bubble';
+
+class ActivityItem extends React.Component {
+  static propTypes = {
+    id: PropTypes.string,
+    date: PropTypes.string,
+    author: PropTypes.shape({
+      type: ActivityAvatar.propTypes.type,
+      user: ActivityAvatar.propTypes.user,
+    }),
+    avatarSize: PropTypes.number,
+
+    // Hides date in header
+    hideDate: PropTypes.bool,
+
+    header: PropTypes.node,
+    footer: PropTypes.node,
+    renderHeader: PropTypes.func,
+    renderFooter: PropTypes.func,
+  };
+
+  render() {
+    const {
+      className,
+      children,
+      avatarSize,
+      id,
+      date,
+      author,
+      hideDate,
+      renderHeader,
+      renderFooter,
+      header,
+      footer,
+    } = this.props;
+
+    const customHeader = typeof renderHeader === 'function';
+    const customFooter = typeof renderFooter === 'function';
+
+    return (
+      <ActivityItemWrapper data-test-id="activity-item" className={className}>
+        {id && <a id={id} />}
+        <StyledActivityAvatar type={author.type} user={author.user} size={avatarSize} />
+
+        <ActivityBubble>
+          {customHeader && renderHeader()}
+
+          {!customHeader && !!header && (
+            <ActivityHeader>
+              <ActivityHeaderContent>{header}</ActivityHeaderContent>
+
+              {!hideDate && date && <StyledTimeSince date={date} />}
+            </ActivityHeader>
+          )}
+
+          {children && typeof children === 'function' && children()}
+          {children && typeof children !== 'function' && (
+            <ActivityBody>{children}</ActivityBody>
+          )}
+
+          {customFooter && renderFooter()}
+          {!customFooter && !!footer && <ActivityFooter>{footer}</ActivityFooter>}
```

Comment:
Rather than have the additional `customFooter` and `customHeader` props, can `footer` and `header` both just accept either a function or a node? This keeps things a bit more consistent with how `children` is being handled above